### PR TITLE
Add k-NN filter to neural search

### LINKED ISSUE #2271: [DOC] Add filter functionality to neural-search docs
**What do you want to do?**
 
- [X] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 Related to https://github.com/opensearch-project/neural-search/issues/86,

The neural query type is a wrapper around the k-NN query type. In #1814 , we introduced filtering for ANN search with lucene for k-NN. Similarly, we have extended this support to the neural query.

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
Please let me know if there are any questions.

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
Signed-off-by: Naarcha-AWS <naarcha@amazon.com>

Fixes #2271.

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _neural-search-plugin/index.md [hunk 1]
```diff
@@ -132,7 +132,9 @@ With the text_embedding processor in place through a Neural Search pipeline, the
 
 ## Search a neural index 
 
-If you want to use a language model to convert a text query into a k-NN vector query, use the `neural` query fields in your query. The neural query request fields can be used in both the [Search API]({{site.url}}{{site.baseurl}}/search-plugins/knn/api/#search-model) and [Query DSL]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/index/). 
+If you want to use a language model to convert a text query into a k-NN vector query, use the `neural` query fields in your query. The neural query request fields can be used in both the [Search API]({{site.url}}{{site.baseurl}}/search-plugins/knn/api/#search-model), [Query DSL]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/index/), and inside a [k-NN search filter]({{site.url}}{{site.baseurl}}/search-plugins/knn/filter-search-knn/).
```
Comment (alicejw-aws) [inline _neural-search-plugin/index.md:135]:
```suggestion
If you want to use a language model to convert a text query into a k-NN vector query, use the `neural` query fields in your query. The neural query request fields can be used in both the [k-NN plugin API]({{site.url}}{{site.baseurl}}/search-plugins/knn/api/#search-model), [Query DSL]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/index/), and inside a [k-NN search filter]({{site.url}}{{site.baseurl}}/search-plugins/knn/filter-search-knn/).
```

Comment (alicejw-aws) [reply _neural-search-plugin/index.md:135]: not sure if you wanted to leave it generically as "Search API" or mention it's the k-NN search API?

**Review by alicejw-aws — APPROVED** [review]
Looks good!
verified link to "Search with k-NN filters"

**Review by kolchfa-aws — APPROVED** [review]
LGTM

**Pull request merged**